core/src/main/java/com/linecorp/armeria/server/metric/PrometheusExpositionService.java

```diff
@@ -85,11 +87,11 @@ public PrometheusExpositionService(CollectorRegistry collectorRegistry) {
 
     @Override
     protected HttpResponse doGet(ServiceRequestContext ctx, HttpRequest req) throws Exception {
-        final ByteArrayOutputStream stream = new ByteArrayOutputStream();
-        try (OutputStreamWriter writer = new OutputStreamWriter(stream)) {
+        final ByteBufOutputStream byteBufOutputStream = new ByteBufOutputStream(ctx.alloc().buffer());
+        try (OutputStreamWriter writer = new OutputStreamWriter(byteBufOutputStream)) {
```

Comment:
I think the writer will close the underlying stream but I would put the bytebuf stream in the try statement too for added clarity.

Reply:
If there is an exception during writing the stream, we need to release the underlying `ByteBuf` too.

Reply:
Thanks fixed.